src/basic/time-util.h

```diff
@@ -29,6 +29,7 @@ typedef struct triple_timestamp {
         usec_t boottime;
 } triple_timestamp;
 
+#define USEC_UNSET    ((usec_t) -2)
```

Comment:
hmm, I'd rather not have more special values for usec_t. I mean, we already had to do usec_add() to deal with USEC_INFINITY, and it becomes nastier if we define more cases like this, as we need to be even more careful then.

Please, just add a separate bool for this stored in the Unit structure that indicates whether the timeout is set or not.

Reply:
Fair enough. I've put it in a structure with a boolean to make the handling a bit more convenient...